Fixes and configuration for pyinotify polling

bryanlandia opened the pull request:
The main fix here is to include IN_CREATE and IN_MOVED_TO to the event mask, so that log rotation of the tracking log doesn't kill the event polling.  

Update to latest release of pyinotify

Add configuration options for pyinotify notifier polling timeout and a sleep delay in times between polls of the log file.

bryanlandia commented:
Hold off reviewing, guys, this actually _doesn't_ fix the problem.

thraxil commented:
@bryanlandia I haven't used `pyinotify` before, but when I've done `inotify` stuff in C, the pattern was to handle `IN_CREATE`/`IN_MOVED_TO` by bailing out and re-registering/initializing everything so it picks up the named file you are interested in again. Unless `pyinotify` is doing some extra magic (totally possible) I think this code is just setting it to treat those events the same as a regular `IN_MODIFY`.

bryanlandia commented:
Thanks @thraxil I had a couple of issues;  for one, IN_CREATE and IN_MOVED_TO only work for watched directories while I'm watching a file.  Second, as you say, I needed to re-initialize everything.  So now I process IN_MOVE_SELF by throwing a custom exception, then catching that to stop my notifier loop and restart my watcher, and it works. cc @thatguynishant

bryanlandia commented:
@thatguynishant I pushed a bunch more commits here.  There were a few problems that I fixed, actually:

* I was trying to exit and restart my `inotify` watcher when the tracking log file moved after log rotation, and was throwing a custom exception to exit the context manager I was using, but I forgot to pass an error message to my custom exception and I was getting a swallowed error that stopped the restart of the watcher
* My http server shutdown code wasn't set up correctly for having the server running in a separate thread
* The code to catch exit signals and run the server shutdown wasn't catching the SIGTERM during a sysctl service restart, so I reimplemented using the signals package

also

* Added a little big more logging
* pyinotify's ProcessEvent code didn't want you to override it's constructor in a subclass, so reworked my subclass a little bit

It seems to be properly handling log rotation now, and restarting properly.  Before, restarts had to keep retrying for 10 seconds or so until the httpd server had shutdown.
